### DISCUSSION TITLE
"Access to requested resource is denied" message for Finances

### DISCUSSION THREAD

**Discussion by yasunori0120** [saleweaver/python-amazon-sp-api#203] [Q&A] [answered]
Hi, 

I wrote a code using python as follows:

response = Orders(marketplace=Marketplaces.JP,credentials=credentials).get_orders(CreatedAfter=(datetime.now() - timedelta(days=5)).isoformat())

This code works fine, but the code

response = Finances(marketplace=Marketplaces.JP,credentials=credentials).list_financial_events(PostedAfter=(datetime.now() - timedelta(days=5)).isoformat())

returns response with "[{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]".

Why does this happen? The method Finances requires different credentials than the method Orders?

**Comment by mscampos92** [comment]
I'm facing the same.. just the orders.get_orders runs... i'm trying to list my catalog products.. and get the same response...

**Reply by vikadis** [reply]
Same issue with me **Order.get_orders** works fine but when using **VendorOrders.get_purchase_orders** with Vendors I get the "Unauthorized" code. Any help here would be greatly appreciated

**Comment by saleweaver** [comment, marked as answer]
Your app in sellercentral needs permission to access the endpoint.
